### ISSUE TITLE
Progress of ad not shown during flash ads?

### ISSUE THREAD

**Issue by nickygerritsen** [MailOnline/videojs-vast-vpaid#135]
I noticed that Flash ads do not show a progress bar. Is this intended?

And is this even fixable or is this too hard?

**Comment by Fetz** [comment]
we can do a loading, but the problem is that some ads they load as well their own stuff so you could have  multiple loading bars

**Comment by nickygerritsen** [comment]
I mean the time-bar with how long the ad takes, not the loading btw :)

**Comment by Fetz** [comment]
Sorry for misunderstanding.

Regarding that it should work, but since is optional some ads doesn't provide the time, then in those cases, we don't show.

http://www.iab.com/wp-content/uploads/2015/06/VPAID_2_0_Final_04-10-2012.pdf
`if is not implemented, the ad unit returns a -1 value. If the duration is unknown, the ad
unit returns a -2. Unknown duration is typical when the user has engaged the ad`

There is 2 events that provide that:
VPAID 1: adRemainingTime
VPAID 2 adDuration

If you have an ad that has this feature implemented and is not working with our videoplayer, please share so we can fix. I will confirm if this feature is really working

**Comment by nickygerritsen** [comment]
It seemed all examples on the demo page that use flash do not work currently

**Comment by Fetz** [comment]
Ok after looking at the code we not handling this in the plugin even if [VPAIDFLASHClient](https://github.com/MailOnline/VPAIDFLASHClient) is triggering this events. I doubt that I can reuse the videojs progress bar

- In the short term I can expose this events to videoplayer so if you want a progress bar you can create yourself.
- later we can do a progress bar with the same styling has videojs.

Let me know if the first solution is ok for you or not

**Comment by nickygerritsen** [comment]
Yeah that's ok as we have our own progress bar for ads already using an ima plugin which we can probably reuse

**Comment by Fetz** [comment]
like the other issue https://github.com/MailOnline/videojs-vast-vpaid/issues/136 if the ad implements is own progress bar you could have a duplicated progress bar

**Comment by nickygerritsen** [comment]
For our case this does not happen at least, so if we get the events we're ok
